src/uwtools/cli.py

```diff
@@ -964,21 +989,34 @@ def _parse_args(raw_args: List[str]) -> Tuple[Args, Checks]:
         STR.template: partial(_add_subparser_template, subparsers),
     }
     drivers = {
-        x: partial(_add_subparser_for_driver, x, subparsers, with_cycle)
-        for x, with_cycle in [
-            (STR.chgrescube, True),
-            (STR.esggrid, False),
-            (STR.fv3, True),
-            (STR.globalequivresol, False),
-            (STR.jedi, True),
-            (STR.makehgrid, False),
-            (STR.mpas, True),
-            (STR.mpasinit, True),
-            (STR.sfcclimogen, False),
-            (STR.ungrib, True),
+        component: partial(_add_subparser_for_driver, component, subparsers)
+        for component in [
+            STR.esggrid,
+            STR.globalequivresol,
+            STR.makehgrid,
+            STR.sfcclimogen,
+        ]
+    }
+    drivers_with_cycle = {
+        component: partial(_add_subparser_for_driver, component, subparsers, with_cycle=True)
+        for component in [
+            STR.chgrescube,
+            STR.fv3,
+            STR.jedi,
+            STR.mpas,
+            STR.mpasinit,
+            STR.ungrib,
+        ]
+    }
+    drivers_with_cycle_and_leadtime = {
+        component: partial(
+            _add_subparser_for_driver, component, subparsers, with_cycle=True, with_leadtime=True
+        )
+        for component in [
+            STR.upp,
         ]
     }
```

Comment:
Slightly longer than the tuple-based version, but possibly clearer in intent, too.